### PR TITLE
LibCore+LibIPC: Quick fixes to get Inspector working

### PR DESCRIPTION
**Pull request by frhun**
Inspector hung on trying to inspect larger applications like WidgetGallery (which I made inspectable by default because it's the perfect target anyways, and who cares about it's performance or security) because the JSON sent to represent the current state by far exceeded (> 200kB) what's possible to be sent in a single write (the fix in EventLoop) and also hit the socket buffer limit and caused an `EAGAIN` error.
The fix for the latter is really far from ideal, but it gets the Inspector working again. Every time it gets triggered a debug message will be written reminding everyone that this problem needs to be addressed, or simply that ludicrously large things are being sent over IPC.

### CODE REVIEW

**Comment by frhun** [comment]
Relates to https://github.com/SerenityOS/serenity/issues/11176

File: Userland/Libraries/LibIPC/Connection.cpp [hunk 1]
```diff
@@ -49,11 +49,20 @@ ErrorOr<void> ConnectionBase::post_message(MessageBuffer buffer)
...
     while (!bytes_to_write.is_empty()) {
         auto maybe_nwritten = m_socket->write(bytes_to_write);
+        writes_done++;
         if (maybe_nwritten.is_error()) {
             auto error = maybe_nwritten.release_error();
             if (error.is_errno()) {
+                // FIXME: This is a hacky way to at least not crash on large messages
+                // The limit of 100 writes is arbitrary, and there to prevent indefinite spinning on the EventLoop
+                if (error.code() == EAGAIN && writes_done < 100) {
+                    sched_yield();
```
Comment (alimpfard) [inline Userland/Libraries/LibIPC/Connection.cpp:63]: Don't you want to yield to the event loop? that'd be `Core::EventLoop::current().pump()`.

Comment (frhun) [reply Userland/Libraries/LibIPC/Connection.cpp:63]: Yes, but actually no. That's the way it should eventually be, but that would probably need extra logic I didn't want to work on now.

Comment (frhun) [reply Userland/Libraries/LibIPC/Connection.cpp:63]: Yes, but actually no.
Something along those lines would be better, but this is basically just a hotfix that works well enough for now.

**Pull request merged**